helper/toggle.go

```diff
@@ -0,0 +1,54 @@
+package helper
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/heroku/color"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"github.com/paketo-buildpacks/libpak/sherpa"
+)
+
+type Toggle struct {
+	Logger bard.Logger
+}
+
+func (t Toggle) Execute() (map[string]string, error) {
+	t.Logger.Infof(color.CyanString("Datadog toggle process start..."))
+	if datadogDisabled(t) {
+		t.Logger.Infof(color.CyanString("Datadog agent disabled by property BPL_DATADOG_DISABLED"))
+		return nil, nil
+	}
+
+	var values []string
+	s, ok := os.LookupEnv("JAVA_TOOL_OPTIONS")
+	if s == "" {
+		return nil, fmt.Errorf("disabling Datadog at launch time is unsupported for Node")
+	}
+	values = append(values, s)
+
+	p, ok := os.LookupEnv("BPI_DATADOG_AGENT_PATH")
+	if !ok {
+		t.Logger.Infof(color.RedString("$BPI_DATADOG_AGENT_PATH is not set during build."))
+		return nil, fmt.Errorf("$BPI_DATADOG_AGENT_PATH must be set")
+	}
+	t.Logger.Infof(color.GreenString("Datadog agent path: %s", p))
+
+	values = append(values, fmt.Sprintf("-javaagent:%s", p))
+	java_tool_options := strings.Join(values, " ")
+
+	return map[string]string{"JAVA_TOOL_OPTIONS": java_tool_options}, nil
+}
+
+func datadogDisabled(t Toggle) bool {
+	val := sherpa.GetEnvWithDefault("BPL_DATADOG_DISABLED", "false")
```

Comment:
You might be able to simply this func with [sherpa.ResolveBool](https://github.com/paketo-buildpacks/libpak/blob/main/sherpa/env_var.go#L57)?
It should allow for various truthy values or return false for unset. Since you wouldn't have to handle an error either you could reduce the func to one line

Reply:
nice, agreed--will change while I'm fixing the test...